documentation/samplefiles-dpg/samplereadme.md

```diff
@@ -0,0 +1,80 @@
+# [[ServiceName]]
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for [[ServiceName]].
+
+## Getting Started
+
+To build the SDKs for My API, simply install AutoRest via `npm` (`npm install -g autorest`) and then run:
+
+> `autorest readme.md`
+
+To see additional help and options, run:
+
+> `autorest --help`
+
+For other options on installation see [Installing AutoRest](https://aka.ms/autorest/install) on the AutoRest github page.
+
+---
+
+## Configuration
+
+### Basic Information
+
+These are the global settings for the [[ServiceName]].
+
+```yaml
+openapi-type: [[OpenApiType]]
+tag: package-[[Version]]
+```
+
+### Security Configuration
+
+These are the global settings for security configuration
+
+> see https://github.com/Azure/autorest/blob/main/docs/generate/authentication.md
+(If authentication is apikey, then the following configuration shall be `security: AzureKey` and `security-header-name: [[HeaderName]]`)
+
+```yaml
+security: AADToken
+security-scopes: [[SecurityScopes]]
+```
+
+
+### Tag: package-[[Version]]
+
+These settings apply only when `--tag=package-[[Version]]` is specified on the command line.
+
+```yaml $(tag) == 'package-[[Version]]'
+input-file:
+  - [[ResourceProviderName]]/[[ReleaseState]]/[[Version]]/[[ServiceName]].json
+```
+
+---
+
+# Code Generation
+
+## Swagger to SDK
+
+This section describes what SDK should be generated by the automatic system.
+This is not used by Autorest itself.
+
+```yaml $(swagger-to-sdk)
+swagger-to-sdk:
+  - repo: azure-sdk-for-python
+  - repo: azure-sdk-for-java
+  - repo: azure-sdk-for-js
+```
+
+## Python
+
+See configuration in [readme.python.md](./readme.python.md)
+
+## TypeScript
+
+See configuration in [readme.typescript.md](./readme.typescript.md)
+
+## Java
+
+See configuration in [readme.java.md](./readme.java.md)
```

Comment:
Why do we need separate files?  Can't we simply provide the configuration inline?

Reply:
It's not suggested to provide the configuration inline because it may makes the `readme.md` big and not friend to read. The `readme.md` is used for common information, and `readme.language.md` is used for language specific information, which is same as mgmt.

Reply:
readmes for DPG should be tiny.  If that is not so then we need to fix the generators.

Reply:
Initially, yes, they should be small.

However, the configure could grow, if customization starts to increase to get in. E.g., when service uses a non-standard LRO, a few lines of configure would be added; when service decide to expose some models directly from swagger (instead of writing it themselves, a few lines would be added).